### PR TITLE
Add `no_std` and basic information about demo slides

### PR DESCRIPTION
**Pull request by JurajSadel**
My motivation for this is to show, what the _minimal_ `no_std` program looks like (instead of scrolling for in the real demo). Also adds some basic information about the demo itself - what it is supposed to do and what external HW/SW we used.

### CODE REVIEW

**Comment by JurajSadel** [comment]
@MabezDev PTAL (for some reason I can't request a review from you :D )

File: espressif-devcon/2023/2023.md [hunk 1]
```diff
@@ -246,6 +246,41 @@ async fn task() {
...
+// Usually, the attribute macro is imported from a crate
+#[entry]
+fn main() -> ! {
+    // Custom initialization logic goes here
+    // This function is the entry point of the program
+    loop {}
+}
+```
+
+# IOT "Hello World" demo
```
Comment (MabezDev) [inline espressif-devcon/2023/2023.md:271]:
```suggestion
# IoT "Hello World" demo
```

File: espressif-devcon/2023/2023.md [hunk 2]
```diff
@@ -246,6 +246,41 @@ async fn task() {
...
+
+What it does:
```
Comment (MabezDev) [inline espressif-devcon/2023/2023.md:273]:
```suggestion
The goal of the demo:
```

Minor nit, feel free to ignore this suggestion.

**Review by MabezDev — CHANGES_REQUESTED** [review]

**Comment by JurajSadel** [comment]
@MabezDev thanks!

**Pull request merged**